src/elements/content-explorer/ContentExplorer.js

```diff
@@ -1431,6 +1433,58 @@ class ContentExplorer extends Component<Props, State> {
         this.setState({ gridColumnCount });
     };
 
+    /**
+     * Function to update metadata field value in metadata based view
+     * @param {BoxItem} item - file item whose metadata is being changed
+     * @param {string} field - metadata template field name
+     * @param {MetadataFieldValue} oldValue - current value
+     * @param {MetadataFieldValue} newVlaue - new value the field to be updated to
+     */
+
+    updateMetadata = (
+        item: BoxItem,
+        field: string,
+        oldValue: ?MetadataFieldValue,
+        newValue: ?MetadataFieldValue,
+    ): void => {
+        this.setState({ isLoading: true });
+        this.metadataQueryAPIHelper.updateMetadata(
+            item,
+            field,
+            oldValue,
+            newValue,
+            () => {
+                this.updateMetadataSuccessCallback(item, field, newValue);
+            },
+            this.errorCallback,
+        );
+    };
+
+    updateMetadataSuccessCallback = (item: BoxItem, field: string, newValue: ?MetadataFieldValue): void => {
+        const { currentCollection }: State = this.state;
+        const { items = [], nextMarker } = currentCollection;
+        const updatedItems = items.map(collectionItem => {
+            const clonedItem = cloneDeep(collectionItem);
+            if (item.id === clonedItem.id) {
+                const fields = getProp(clonedItem, 'metadata.enterprise.fields', []);
+                fields.forEach(itemField => {
+                    if (itemField.name === field) {
+                        itemField.value = newValue; // set updated metadata value to correct item in currentCollection
+                    }
+                });
+            }
+            return clonedItem;
+        });
+
+        this.setState({
+            currentCollection: {
+                items: updatedItems,
+                nextMarker,
+                percentLoaded: 100,
```

Comment:
do you also need to reset isLoading state to false, or is updating percentLoaded sufficient?

Reply:
Looks like I don't need `isLoading`. This flag is used to control dialog(modal) behavior. `percentLoaded` is sufficient.